cmake/cares.cmake

```diff
@@ -28,10 +28,6 @@ if(gRPC_CARES_PROVIDER STREQUAL "module")
     set(_gRPC_CARES_LIBRARIES c-ares)
   endif()
 
-  if(gRPC_INSTALL)
-    message(WARNING "gRPC_INSTALL will be forced to FALSE because gRPC_CARES_PROVIDER is \"module\"")
```

Comment:
Two concerns
- I remember this check was added because we had trouble making the installation work when importing dependencies via add_subdirectory() (see https://github.com/grpc/grpc/pull/11140/files)
- qq: If you install with the dependencies added via add_subdirectory(), isn't that an antipattern? It means that all the dependencies will be installed as part of gRPC (statically linked). Besides other things, it can influence the correctness of e.g. our pkgconfig files?

Reply:
> * isn't that an antipattern?

I wouldn't necessarily say so. A "batteries included" installation can be much more convenient for users.

> * Besides other things, it can influence the correctness of e.g. our pkgconfig files?

To test this out, I locally modified the pkgconfig distribtest to install a newer version of CMake and do an install-from-submodule. "helloworld" still built fine using the resulting pkgconfig files.